.travis.yml

```diff
@@ -0,0 +1,109 @@
+
+#=============================================================================
+# Environment
+#=============================================================================
+# Use new Travis infrastructure (Docker can't sudo yet)
+sudo: false
+language: cpp
+#PYTHON_VERSION=3.6.7
+
+#=============================================================================
+# Build matrix
+#=============================================================================
+matrix:
+  fast_finish: true
+  include:
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=ON"
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=OFF"
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=ON"
+    - os: linux
+      dist: xenial
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=OFF"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=ON"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_MPI=OFF"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=ON"
+    - os: osx
+      osx_image: xcode10.2
+      env:
+        - cmake_option="-DCORENRN_ENABLE_SOA=OFF"
+#=============================================================================
+# Common Packages
+#=============================================================================
+addons:
+  # for Linux builds, we use APT
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+    packages:
+      - flex
+      - bison
+      - libopenmpi-dev
+      - openmpi-bin
+#      - cmake  #cmake 3.12.4 on travis
+#      - libboost-all-dev
+#      - python3-dev
+#      - python3-pip
+
+  # for Mac builds, we use Homebrew
+  homebrew:
+    packages:
+      - flex
+      - bison
+      - openmpi
+#      - cmake  #cmake 3.11.3 on travis
+#      - boost
+#      - python@3
+
+#=============================================================================
+# Install dependencies : flex
+#=============================================================================
+before_install:
+  # brew installed flex and bison is not in $PATH
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    export PATH=/usr/local/opt/flex/bin:/usr/local/opt/bison/bin:$PATH;
+    else
+    pyenv global $PYTHON_VERSION;
+    ls /usr/bin/;
+    fi
+
+#=============================================================================
+# Build, test and install
+#=============================================================================
+script:
+  - echo "------- Build, Test and Install -------"
+  - mkdir build && cd build
+  - cmake ${cmake_option} -DCMAKE_INSTALL_PREFIX=$HOME/CoreNeuron ..
+  - cmake --build .
+  - ctest --output-on-failure
+  - make install
+
+#=============================================================================
```

Comment:
Please remove the notifications section if unused

Reply:
Thanks!. I have implemented your comments, removed deadcode and unneeded lines.